File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.cpp
class Solution {
public:
    int i = 0;
    string reverseParentheses(string s) {
        string answer = "";
        for(;i < s.size() && s[i] != ')'; i++) {
            if(s[i] == '(') {
                i++;
                answer += this->reverseParentheses(s);
            } else {
                answer += s[i];
            }
        }
        
        if(i < s.size()) reverse(answer.begin(), answer.end());
        return answer;
    }
};

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.java
class Solution {
    int i;
    public String reverseParentheses(String s) {
        String answer = "";
        for(;i < s.length() && s.charAt(i) != ')'; i++) {
            if(s.charAt(i) == '(') {
                i++;
                answer += reverseParentheses(s);
            } else {
                answer += s.charAt(i);
            }
        }
        
        if(s.length() != i) {
            String tmp = "";
            for(char c : answer.toCharArray()) {
                tmp = c + tmp;
            }
            
            return tmp;
        }
        
        return answer;
    }
}